ui/src/ducks/inventory/index.js

```diff
@@ -1,17 +1,110 @@
+import axios from 'axios'
 import { createAction, handleActions } from 'redux-actions'
+import { openError, openSuccess } from '../alerts'
 
+
+//action types
 const actions = {
   INVENTORY_GET_ALL: 'inventory/get_all',
+  INVENTORY_GET_ALL_PENDING: 'inventory/get_all_PENDING',
+  INVENTORY_DELETE: 'inventory/delete',
+  INVENTORY_CREATE: 'inventory/create',
+  INVENTORY_UPDATE: 'inventory/update',
+  INVENTORY_REFRESH: 'inventory/refresh'
 }
 
 export let defaultState = {
-  all: []
+  all: [],
+  fetched: false,
 }
 
-export const findInventory = createAction(actions.INVENTORY_GET_ALL, () => {
-  //TODO
-})
+//action creator
+export const findInventory = createAction(actions.INVENTORY_GET_ALL, () =>
+  (dispatch, getState, config) => axios
+    .get(`${config.restAPIUrl}/inventory`)
+    .then((suc) => {
+      dispatch(refreshInventory(suc.data))
+    })
+)
+
+export const refreshInventory = createAction(actions.INVENTORY_REFRESH, (payload) =>
+  (dispatcher, getState, config) =>
+    payload.sort((inventoryA, inventoryB) =>
+      inventoryA.name < inventoryB.name ? -1 : inventoryA.name > inventoryB.name ? 1 : 0)
+)
+
+export const createInventory = createAction(actions.INVENTORY_CREATE, (payload) =>
+  (dispatch, getState, config) => axios
+    .post(`${config.restAPIUrl}/inventory`, payload)
+    .then((suc) => {
+      //add newly created inventory to current state's list of objects.
+      const invs = []
+      getState().inventory.all.forEach(inv => {
+        if (inv.id !== suc.data.id) {
+          invs.push(inv)
+        }
+      })
+      invs.push(suc.data)
+      dispatch(refreshInventory(invs))
+      const succMsg = 'Inventory successfully created.'
+      dispatch(openSuccess(succMsg))
+    })
+    .catch((error) => {
+      const errMsg = 'Failed to create Inventory.'
+      dispatch(openError(errMsg))
+    })
+)
+
+export const updateInventory = createAction(actions.INVENTORY_UPDATE, (payload) =>
+  (dispatch, getState, config) => axios
+    .post(`${config.restAPIUrl}/inventory/update`, payload)
```

Comment:
change this to .put Forgot to add this feedback in my original requested changes